### DISCUSSION TITLE
Pod Crash

### DISCUSSION THREAD

**Discussion by userH1213** [InterDigitalInc/AdvantEDGE#355] [Q&A] [answered]
Hello everyone,

as you can see in the image below I have a pod that is failing .. how can I fix it please..
![meepPodsfail](https://user-images.githubusercontent.com/79055649/162063900-d5b24c24-419b-4f3b-9188-3fa8a8fccad8.png)

outputs that someone may need based on similar issues I read here.
![helmForFailingPod](https://user-images.githubusercontent.com/79055649/162131682-158930ff-a107-449c-aa1d-37467da3e11a.png)

admin1@admin1:~/AdvantEDGE/go-apps/meepctl$ meepctl version all
Using repo config file: /home/admin1/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/admin1/.meepctl.yaml
{
  "name": "meepctl",
  "version": "1.8.1"
}
{
  "name": ".meepctl-repocfg.yaml",
  "version": "1.8.1"
}
{
  "name": "meep-cert-manager",
  "version": "NA"
}
{
  "name": "couchdb",
  "version": "couchdb:3.1.0",
  "id": "b604d056d8024f10346eab768de7aea06bc0a1b7c55d6087e1b1cd4328c8061c"
}
{
  "name": "docker-registry",
  "version": "registry:2.7.1",
  "id": "169211e20e2f2d5d115674681eb79d21a217b296b43374b8e39f97fcf866b375"
}
{
  "name": "grafana",
  "version": "grafana/grafana:7.3.5",
  "id": "511bc20bfcd1b79f3947bb1c33d152f7484e7a91418883fb4dddf71274227321"
}
{
  "name": "meep-influxdb",
  "version": "influxdb:1.8.0-alpine",
  "id": "5eca9dfe9930a3325323cef801827eb1b0940070465f8f215447b8e732c72b34"
}
{
  "name": "meep-ingress",
  "version": "NA"
}
{
  "name": "kube-state-metrics",
  "version": "quay.io/coreos/kube-state-metrics:v1.9.7",
  "id": "2f82f0da199c60a7699c43c63a295c44e673242de0b7ee1b17c2d5a23bec34cb"
}
{
  "name": "meep-minio",
  "version": "NA"
}
{
  "name": "meep-open-map-tiles",
  "version": "NA"
}
{
  "name": "meep-postgis",
  "version": "postgis/postgis:12-3.0",
  "id": "71acda16357f2973034483a4a8363cc9499061120b592bcc3b7f2fbed82da621"
}
{
  "name": "meep-prometheus",
  "version": "NA"
}
{
  "name": "meep-redis",
  "version": "NA"
}
{
  "name": "meep-thanos",
  "version": "NA"
}
{
  "name": "meep-thanos-archive",
  "version": "NA"
}
{
  "name": "helm",
  "version": "v3.3.4",
  "id": "a61ce5633af99708171414353ed49547cf05013d"
}
{
  "name": "docker client",
  "version": "20.10.14",
  "id": "a224086"
}
{
  "name": "docker server",
  "id": "go1.16.15"
}
{
  "name": "k8s client",
  "version": "v1.19.1",
  "id": "206bcadf021e76c27513500ca24182692aabd17e"
}
{
  "name": "k8s server",
  "version": "v1.19.16",
  "id": "e37e4ab4cc8dcda84f1344dda47a97bb1927d074"
}
{
  "name": "weave",
  "version": "ghcr.io/weaveworks/launcher/weave-kube:2.8.1",
  "id": "d797338e7beb17222e10757b71400d8471bdbd9be13b5da38ce2ebf597fb4e63"
}
{
  "name": "meep-auth-svc",
  "version": "meep-docker-registry:30001/meep-auth-svc:latest",
  "id": "0601dc4c821afe832590d1ab9c3be539ab3cf4f7a247b432a652671292ee5cc1",
  "build": "sha256sum:"
}
{
  "name": "meep-ingress-certs",
  "version": "NA"
}
{
  "name": "meep-mon-engine",
  "version": "meep-docker-registry:30001/meep-mon-engine:latest",
  "id": "b418dfd8d21d512138c53188961fa8a89f219041ffc54e509aba69a1d90a387d",
  "build": "sha256sum:"
}
{
  "name": "meep-platform-ctrl",
  "version": "meep-docker-registry:30001/meep-platform-ctrl:latest",
  "id": "f34889d5e4fdb333511c7b8b132fde70ecbc12f4639606e7dd2b33b1a7585352",
  "build": "sha256sum:"
}
{
  "name": "meep-virt-engine",
  "version": "meep-docker-registry:30001/meep-virt-engine:latest",
  "id": "a1bd84802687b75f6ef52ecf3a607b613c4223b1a4484f4f92b50a85b1d44f3d",
  "build": "73a36aab073b7b03ff23b93945cd791660b108ae2d3c0159803594eb232d7682"
}
{
  "name": "meep-webhook",
  "version": "meep-docker-registry:30001/meep-webhook:latest",
  "id": "d8bb49fa18c1979926dec5f332a305f0e8554522212f44e754d62495e3706f62",
  "build": "b1f550c998b30e1bf0925c22bb9451d747c748e8f07a379ebd908e47a2c2095a"
}
admin1@admin1:~/AdvantEDGE/go-apps/meepctl$ git log --oneline --decorate --graph
fatal: not a git repository (or any of the parent directories): .git
admin1@admin1:~/AdvantEDGE/go-apps/meepctl$ kubectl describe pods meep-ingress-controller-rmt4x
Name:         meep-ingress-controller-rmt4x
Namespace:    default
Priority:     0
Node:         admin1/192.168.0.125
Start Time:   Sun, 03 Apr 2022 05:47:54 -0400
Labels:       app=nginx-ingress
              app.kubernetes.io/component=controller
              app.kubernetes.io/instance=meep-ingress
              app.kubernetes.io/name=ingress-nginx
              component=controller
              controller-revision-hash=546b7d9d4c
              pod-template-generation=1
              release=meep-ingress
Annotations:  <none>
Status:       Running
IP:           192.168.0.125
IPs:
  IP:           192.168.0.125
Controlled By:  DaemonSet/meep-ingress-controller
Containers:
  controller:
    Container ID:  docker://2647c1338ce53047ebd7a101c51feb1440d6248fedeb2f94320b15ad8dafb1c0
    Image:         k8s.gcr.io/ingress-nginx/controller:v0.44.0@sha256:3dd0fac48073beaca2d67a78c746c7593f9c575168a17139a9955a82c63c4b9a
    Image ID:      docker-pullable://k8s.gcr.io/ingress-nginx/controller@sha256:3dd0fac48073beaca2d67a78c746c7593f9c575168a17139a9955a82c63c4b9a
    Ports:         80/TCP, 443/TCP, 10254/TCP
    Host Ports:    80/TCP, 443/TCP, 10254/TCP
    Args:
      /nginx-ingress-controller
      --http-port=80
      --https-port=443
      --default-backend-service=default/meep-ingress-defaultbackend
      --election-id=ingress-controller-leader
      --ingress-class=nginx
      --configmap=default/meep-ingress-controller
      --default-ssl-certificate=default/meep-ingress
      --metrics-per-host=false
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Thu, 07 Apr 2022 01:49:24 -0400
      Finished:     Thu, 07 Apr 2022 01:49:25 -0400
    Ready:          False
    Restart Count:  66
    Requests:
      cpu:      100m
      memory:   90Mi
    Liveness:   http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=5
    Readiness:  http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=3
    Environment:
      POD_NAME:       meep-ingress-controller-rmt4x (v1:metadata.name)
      POD_NAMESPACE:  default (v1:metadata.namespace)
      LD_PRELOAD:     /usr/local/lib/libmimalloc.so
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from meep-ingress-token-2lxtr (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  meep-ingress-token-2lxtr:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  meep-ingress-token-2lxtr
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  kubernetes.io/os=linux
Tolerations:     node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                 node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                 node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                 node.kubernetes.io/not-ready:NoExecute op=Exists
                 node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                 node.kubernetes.io/unreachable:NoExecute op=Exists
                 node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason   Age                  From             Message
  ----     ------   ----                 ----             -------
  Normal   Created  9h (x17 over 10h)    kubelet, admin1  Created container controller
  Warning  BackOff  23s (x559 over 10h)  kubelet, admin1  Back-off restarting failed container
admin1@admin1:~/AdvantEDGE/go-apps/meepctl$ kubectl logs meep-ingress-controller-rmt4x
-------------------------------------------------------------------------------
NGINX Ingress controller
  Release:       v0.44.0
  Build:         f802554ccfadf828f7eb6d3f9a9333686706d613
  Repository:    https://github.com/kubernetes/ingress-nginx
  nginx version: nginx/1.19.6

-------------------------------------------------------------------------------

I0407 05:54:29.328795       7 flags.go:208] "Watching for Ingress" class="nginx"
W0407 05:54:29.328940       7 flags.go:213] Ingresses with an empty class will also be processed by this Ingress controller
F0407 05:54:29.329411       7 main.go:68] port 80 is already in use. Please check the flag --http-port
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc000138001, 0xc000020000, 0x67, 0xa4)
	k8s.io/klog/v2@v2.4.0/klog.go:1026 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x26915e0, 0xc000000003, 0x0, 0x0, 0xc000464700, 0x25e6c33, 0x7, 0x44, 0x40e200)
	k8s.io/klog/v2@v2.4.0/klog.go:975 +0x19b
k8s.io/klog/v2.(*loggingT).printDepth(0x26915e0, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc0004566b0, 0x1, 0x1)
	k8s.io/klog/v2@v2.4.0/klog.go:732 +0x16f
k8s.io/klog/v2.(*loggingT).print(...)
	k8s.io/klog/v2@v2.4.0/klog.go:714
k8s.io/klog/v2.Fatal(...)
	k8s.io/klog/v2@v2.4.0/klog.go:1482
main.main()
	k8s.io/ingress-nginx/cmd/nginx/main.go:68 +0x1fe

goroutine 18 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x26915e0)
	k8s.io/klog/v2@v2.4.0/klog.go:1169 +0x8b
created by k8s.io/klog/v2.init.0
	k8s.io/klog/v2@v2.4.0/klog.go:417 +0xdf

Thank you

**Comment by dilallkx** [comment, marked as answer]
Hi @userH1213,

Thank you for reaching out.

From your logs, it seems you have a service already running on port 80 on your host:
`F0407 05:54:29.329411 7 main.go:68] port 80 is already in use. Please check the flag --http-port`

The ingress controller is configured to use ports 80/443 by default. You can either stop the service that is using port 80 to free up the port, or configure the AdvantEDGE platform deployment to use different ports. You can do so in the repocfg file by updating the values for `repo.deployment.ingress.http-port`.

Let me know if this resolves your issue.
Regards!

**Reply by userH1213** [reply]
Hi,
I made sure that port 80 is free. but it didn't resolve the issue. I deleted the dependencies and the core, then I deployed the dependencies, dockerize all and deployed the core. still didn't work.

![port80Listening](https://user-images.githubusercontent.com/79055649/162576646-b00aee43-0435-4bb6-9a49-349fb3503a1c.png)

Here is the description 

admin1@admin1:~/AdvantEDGE/go-apps/meepctl$ kubectl describe pods meep-ingress-controller-h482v
Name:         meep-ingress-controller-h482v
Namespace:    default
Priority:     0
Node:         admin1/192.168.0.125
Start Time:   Sat, 09 Apr 2022 08:48:22 -0400
Labels:       app=nginx-ingress
              app.kubernetes.io/component=controller
              app.kubernetes.io/instance=meep-ingress
              app.kubernetes.io/name=ingress-nginx
              component=controller
              controller-revision-hash=546b7d9d4c
              pod-template-generation=1
              release=meep-ingress
Annotations:  <none>
Status:       Running
IP:           192.168.0.125
IPs:
  IP:           192.168.0.125
Controlled By:  DaemonSet/meep-ingress-controller
Containers:
  controller:
    Container ID:  docker://fa29ebac72581b46e6f2f50dd75d42c30562263c6d5bfd28b33cd13ceb55d8a6
    Image:         k8s.gcr.io/ingress-nginx/controller:v0.44.0@sha256:3dd0fac48073beaca2d67a78c746c7593f9c575168a17139a9955a82c63c4b9a
    Image ID:      docker-pullable://k8s.gcr.io/ingress-nginx/controller@sha256:3dd0fac48073beaca2d67a78c746c7593f9c575168a17139a9955a82c63c4b9a
    Ports:         80/TCP, 443/TCP, 10254/TCP
    Host Ports:    80/TCP, 443/TCP, 10254/TCP
    Args:
      /nginx-ingress-controller
      --http-port=80
      --https-port=443
      --default-backend-service=default/meep-ingress-defaultbackend
      --election-id=ingress-controller-leader
      --ingress-class=nginx
      --configmap=default/meep-ingress-controller
      --default-ssl-certificate=default/meep-ingress
      --metrics-per-host=false
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Sat, 09 Apr 2022 09:16:58 -0400
      Finished:     Sat, 09 Apr 2022 09:16:59 -0400
    Ready:          False
    Restart Count:  10
    Requests:
      cpu:      100m
      memory:   90Mi
    Liveness:   http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=5
    Readiness:  http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=3
    Environment:
      POD_NAME:       meep-ingress-controller-h482v (v1:metadata.name)
      POD_NAMESPACE:  default (v1:metadata.namespace)
      LD_PRELOAD:     /usr/local/lib/libmimalloc.so
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from meep-ingress-token-8tb5h (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  meep-ingress-token-8tb5h:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  meep-ingress-token-8tb5h
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  kubernetes.io/os=linux
Tolerations:     node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                 node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                 node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                 node.kubernetes.io/not-ready:NoExecute op=Exists
                 node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                 node.kubernetes.io/unreachable:NoExecute op=Exists
                 node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  31m                  default-scheduler  Successfully assigned default/meep-ingress-controller-h482v to admin1
  Warning  Unhealthy  30m (x2 over 31m)    kubelet, admin1    Readiness probe failed: Get "http://192.168.0.125:10254/healthz": dial tcp 192.168.0.125:10254: connect: connection refused
  Warning  Unhealthy  30m (x2 over 31m)    kubelet, admin1    Liveness probe failed: Get "http://192.168.0.125:10254/healthz": dial tcp 192.168.0.125:10254: connect: connection refused
  Normal   Pulled     29m (x4 over 31m)    kubelet, admin1    Container image "k8s.gcr.io/ingress-nginx/controller:v0.44.0@sha256:3dd0fac48073beaca2d67a78c746c7593f9c575168a17139a9955a82c63c4b9a" already present on machine
  Normal   Created    29m (x4 over 31m)    kubelet, admin1    Created container controller
  Normal   Started    29m (x4 over 31m)    kubelet, admin1    Started container controller
  Warning  BackOff    91s (x139 over 30m)  kubelet, admin1    Back-off restarting failed container



Thank you for your help

**Reply by userH1213** [reply]
running kubectl get pods --all-namespaces | grep meep   few minutes later
![Screenshot from 2022-04-09 09-46-33](https://user-images.githubusercontent.com/79055649/162577081-f825b38a-20bb-4453-a92a-48fe9678346f.png)

**Reply by dilallkx** [reply]
Hi @userH1213,

Could you please provide the logs from before the crash using the following command:
`kubectl logs -p <pod-name>`

Also, please provide output for the following command:
`kubectl describe node`

Hopefully this can help us identify the crash cause.
Regards!

**Reply by userH1213** [reply]
Hi,
To be honest I wasn't sure what you meant by the logs before the crash, I don't know how to determine exactly when the crash happened. So I deleted the core and dep again and reinstall them again, while doing that I tried to check the commands you provided.

![kubectlLogs](https://user-images.githubusercontent.com/79055649/163342029-f8b7f4ae-79e7-490f-8780-67fab8534f08.png)

![kubectlDescribeNode1](https://user-images.githubusercontent.com/79055649/163342063-8eea2df0-6cf5-417e-bfa2-2ac0804e2e30.png)

![kubectlDescribeNode2](https://user-images.githubusercontent.com/79055649/163342088-ca0e20ab-3de7-48bc-9e72-11f72ac34177.png)

I also noticed that there is an issue with importing the C.
![couldnotImportCError](https://user-images.githubusercontent.com/79055649/163342677-a2ad629d-4fae-41de-bc6f-217cd4054565.png)

![importCNotAllowed](https://user-images.githubusercontent.com/79055649/163342702-9fe954a9-80b9-4673-bef4-a1112365937c.png)

I'm really grateful for your help
Thank you

**Reply by dilallkx** [reply]
Hi @userH1213,

Your first screenshot shows that port 80 is already in use and is conflicting with nginx-ingress. You seem to have an apache server running. Please try stopping it and then redeploy AdvantEDGE. If it still fails, please provide the failure logs for the ingress controller.

For the meepctl lint issue, please verify that you have the correct golangci-lint version installed, as specified in the doc here:
https://interdigitalinc.github.io/AdvantEDGE/docs/setup/env-dev/#golangci-lint
If the problem persists, please create a new discussion for this issue.

Regards!

**Reply by userH1213** [reply]
It's working now!
![stoppingApache2](https://user-images.githubusercontent.com/79055649/163702284-ece73c35-bdc9-4178-b487-fff29d27bfd3.png)

![showingWorkingmeepctl](https://user-images.githubusercontent.com/79055649/163702289-56340348-f757-464d-bec9-f44808e74e7c.png)

I'll start working on the demos now.
Again Thank you for your help.

**Reply by dilallkx** [reply]
Great to hear your issue has been resolved!
Best regards!
